pilot/pkg/networking/core/v1alpha3/gateway.go

```diff
@@ -40,6 +40,14 @@ import (
 	"istio.io/pkg/log"
 )
 
+type workloadSdsConfig struct {
```

Comment:
it's a bit of weird to define `workloadSdsConfig` just in gateway.go

I mean within-mesh workload listener/cluster, we need same information as well? How are things handled there? Can't we use the same struct?

Reply:
Agreed that `cluster.go` is a super awkward place to define this. In all other places, the relevant `SDS` fields are pulled out [straight from `env.Mesh`](https://github.com/istio/istio/blob/a696cf4dc23eae4cad54719735314e0396a9f433/pilot/pkg/networking/core/v1alpha3/cluster.go#L1046). Here, we'd have to thread those args down through a few call layers, so it seemed cleaner to bundle them in a struct.

wdyt? now that I've thought more about it, it would probably make more sense just to thread them down.

Reply:
Having a struct is fine, might worth adding a struct to `pilot/pkg/security/model` file, and later refactor the cluster/listeners' inlined SDS config as well?

@quanjielin can you comment on this as well?